### DISCUSSION TITLE
Connection to csv, txt and xlsx

### DISCUSSION THREAD

**Discussion by PedroReus** [dbeaver/dbeaver#14637] [Q&A]
Dear

Along with saying hello, I present the following query, I try to import a file in csv or text or xlsx format, but I don't know where I should connect and how the connection is managed in relation to the configuration

**Comment by LonwoLonwo** [comment]
Hello @PedroReus 

If I understand you correctly, do you have the file with some data? What kind of data it is? Any database-specific fields (like timestamp with timezone)? Maybe you know, what database was used when creating this file?

**Comment by PedroReus** [comment]
Hello, I actually have files in csv, txt and xlsx format which I need to import in order to join. The data I have is of type number, text and dates.
Therefore I need to know how to connect and configure the import of these files to DBeaver correctly.
It would be helpful if you can show me the step by step or coordinate a video call to know what I should do

**Comment by LonwoLonwo** [comment]
Ok. We have much helpful info in our Wiki: https://github.com/dbeaver/dbeaver/wiki

You can read about the new connection here: https://github.com/dbeaver/dbeaver/wiki/Create-Connection
About data import here: https://github.com/dbeaver/dbeaver/wiki/Data-transfer#importing-data-from-csv-format

Also, you can add the folder with your CSV (only CSV) files as a separate connection.

![2021-11-19 10_16_32-Connect to a database](https://user-images.githubusercontent.com/45152336/142581298-83bdc600-c8a9-4151-9a41-bd59f12e8372.png)

With CSV driver you can open SQL Editor and create SELECT queries with JOINs: https://github.com/dbeaver/dbeaver/wiki/SQL-Editor
But you can't change data, just read.

**Comment by PedroReus** [comment]
I've been looking at what you wrote me yet I can't connect to a csv or txt file.
What I do is the following
1) I select the connector
![image](https://user-images.githubusercontent.com/91746607/142616501-16909365-739f-49e2-a73e-7fee50b77b54.png)
2) I get the following configuration, I understand that I must look for the path of the files in csv format
![image](https://user-images.githubusercontent.com/91746607/142616537-61fa25f2-4d2d-4ddf-be35-d5ceca45a41c.png)
3) I look for the path of my files and finish the task
![image](https://user-images.githubusercontent.com/91746607/142616558-0074d2df-1692-406a-9f5f-7858d414340c.png)
4) This appears as a connector
![image](https://user-images.githubusercontent.com/91746607/142616582-1e042f3b-6d6f-4dba-b802-9d9e90c10ea0.png)
5) I get this error
![image](https://user-images.githubusercontent.com/91746607/142616619-105c46d9-fc28-4b71-8b15-4568b2bc25bb.png)


I understand that I have to configure some drivers but I don't know what it is or if I'm doing something wrong

**Comment by LonwoLonwo** [comment]
Maybe you have some connection restriction for DBeaver, I don't know.

You can add download and add driver manually in the Connection Settings -> Edit Driver Settings -> Libraries -> Add File/Download

![2021-11-19 18_32_57-DBeaver 21 1 0 - system 2](https://user-images.githubusercontent.com/45152336/142649286-cd12af9e-c3ad-4066-8e2d-87f476a82384.png)

![2021-11-19 18_33_43-Connect to a database](https://user-images.githubusercontent.com/45152336/142649288-caa27069-c368-45b5-80aa-cf8fd121af13.png)

But I see that you have many Oracle connections.
You can import your files into the new tables.

![2021-11-19 18_38_12-](https://user-images.githubusercontent.com/45152336/142649660-7d298057-fe0c-4355-b2aa-bb4df544ce30.png)

**Comment by PedroReus** [comment]
I was able to connect with the csv format. (Thanks for the support).

Now I don't know if DBeader can do the same for files (tables) in xlsx or txt format?

**Comment by LonwoLonwo** [comment]
> Now I don't know if DBeader can do the same for files (tables) in xlsx or txt format?

DBeaver can not open XLSX or txt files as tables. Only CSV.

Also, the ability to use import from XLSX file you can use only in [DBeaver commercial version](https://dbeaver.com/).

**Comment by p91paul** [comment]
@PedroReus for txt, it depends on the content. If the txt is actually similar to a csv (fields delimited by something) the csv connection lets you specify delimiter and extension.

**Comment by PedroReus** [comment]
@p91paul Perfect, and in relation to the excel (xlsx) format, could the csv connection be used?

**Reply by p91paul** [reply]
I think for xlsx what @LonwoLonwo said applies. You cannot open them as tables but you can import them on the enterprise version. The csv driver only supports delimited text files.

**Comment by TiViAx** [comment]
I'm giving you the answer late, but you can save the xlsx file in csv format, although each individual sheet must be saved to a separate csv file. This way you will get a database of n-files. Next, we do all of the above.
